lib/ttnt/testtask.rb

```diff
@@ -0,0 +1,35 @@
+module TTNT
+  class TestTask
+    @@instances = []
+
+    def self.instances
+      @@instances
+    end
+
+    ATTRIBUTES = %i(
+      name
+      libs
+      pattern
+      options
+      test_files
+      warning
+      loader
+      ruby_opts
+      description
+      pattern
+    ).freeze
+
+    ATTRIBUTES.each do |attr|
+      attr_reader attr
+    end
+
+    def initialize(rake_test_task)
+      ATTRIBUTES.each do |attr|
```

Comment:
What do you think about rewriting this like:

~~~ruby
def initialize(test_task)
  attributes = test_task.instance_variables - [:@test_files]
  attributes.map! { |attribute| attribute[1..-1] }

  attributes.each do |ivar|
    self.class.class_eval("attr_accessor :#{ivar}")
    send(:"#{ivar}=", test_task.send(:"#{ivar.to_s}"))
  end

  @@instances << self
end
~~~

Thus you don't have to maintain the attribute's array since it can change between versions and you put `pattern` two times while `verbose` is missing. :smile:The `[1..-1]` is just to remove the "@" at the beginning of the symbol.

I don't know what the `respond_to?` call is for ; maybe the missing `test_files` attribute reader ?